dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -11,7 +11,7 @@ class OmniauthCallbacksController < Devise::OmniauthCallbacksController
   # TODO: figure out how to avoid skipping CSRF verification for Powerschool
   skip_before_action :verify_authenticity_token, only: :powerschool
 
-  before_action :check_account_linking_lock, on: %i[connect_provider link_accounts]
```

Comment:
Note that `on:` is actually intended to be `only:`. However, because `on:` is not a valid key, this `before_action` was happening for all controller actions. That's why  `check_account_linking_lock` was called when the user was simply logging in with OAuth.